From: https://patch-diff.githubusercontent.com/raw/MariaDB/server/pull/2006.patch
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Tue, 8 Feb 2022 00:18:27 +0000
Subject: [PATCH] MDEV-27804 Fails to build - perf schema - thread id of type
 uintptr_t requires header

While building on GNU/Hurd and kfreebsd.

On the C++ standard uintptr_t can be defined in <cstdint>
ref: https://www.cplusplus.com/reference/cstdint/

Fixes: 0d44792a835128a83ff07f14dbbcdd621df9f7da
---
 storage/perfschema/my_thread.h | 4 ++++
 1 file changed, 4 insertions(+)

--- a/storage/perfschema/my_thread.h
+++ b/storage/perfschema/my_thread.h
@@ -14,6 +14,10 @@
 #include <pthread_np.h>
 #endif
 
+#if defined(HAVE_INTEGER_PTHREAD_SELF)
+#include <cstdint>
+#endif
+
 typedef pthread_key_t thread_local_key_t;
 typedef pthread_t my_thread_handle;
 typedef pthread_attr_t my_thread_attr_t;
